src/libstd/env.rs

```diff
@@ -182,6 +182,12 @@ impl fmt::Debug for VarsOs {
 /// * Environment variable is not present
 /// * Environment variable is not valid unicode
 ///
+/// # Panics
+///
+/// This function may panic if `key` is empty, contains an ASCII equals sign
```

Comment:
Is it "may panic" or "panics" (definitive)?

Reply:
The function relies on `std::sys::os`, so the behaviour differs by operating system. We could update the docs to make this clearer.

Reply:
Hmm, I'm not sure what is the practice in this case. Maybe "may" is fine.

BTW, I do wonder why this behavior was chosen, that if `key` contains a nul then it panics. It makes sense for *setting* a var, but when getting, it makes more sense to me that it just return `None`. The code comment in the unix implementation seems to agree, I think..

https://github.com/rust-lang/rust/blob/ad7c55e1fc55d9af4787b285cec1c64e3480ae84/src/libstd/sys/unix/os.rs#L469-L471